Updated pre-commit hooks

yunchu opened the pull request:
<!-- Contributing guide: https://github.com/openvinotoolkit/training_extensions/blob/develop/CONTRIBUTING.md -->

### Summary
- Enabled 'ruff' as a base linter
- Fixed some issues reported by 'ruff' and 'mypy'
- Clean up some outdated configurations in pre-commit-config
- Removed unused config files

<!--
Resolves #111 and #222.
Depends on #1000 (for series of dependent commits).

This PR introduces this capability to make the project better in this and that.

- Added this feature
- Removed that feature
- Fixed the problem #1234
-->

### How to test

<!-- Describe the testing procedure for reviewers, if changes are
not fully covered by unit tests or manual testing can be complicated. -->

### Checklist

<!-- Put an 'x' in all the boxes that apply -->

- [ ] I have added unit tests to cover my changes.​
- [ ] I have added integration tests to cover my changes.​
- [ ] I have added e2e tests for validation.
- [ ] I have added the description of my changes into CHANGELOG in my target branch (e.g., [CHANGELOG](https://github.com/openvinotoolkit/training_extensions/blob/develop/CHANGELOG.md) in develop).​
- [ ] I have updated the documentation in my target branch accordingly (e.g., [documentation](https://github.com/openvinotoolkit/training_extensions/tree/develop/docs) in develop).
- [ ] I have [linked related issues](https://help.github.com/en/github/managing-your-work-on-github/linking-a-pull-request-to-an-issue#linking-a-pull-request-to-an-issue-using-a-keyword).

### License

- [ ] I submit _my code changes_ under the same [Apache License](https://github.com/openvinotoolkit/training_extensions/blob/develop/LICENSE) that covers the project.
      Feel free to contact the maintainers if that's a concern.
- [ ] I have updated the license header for each file (see an example below).

```python
# Copyright (C) 2023 Intel Corporation
# SPDX-License-Identifier: Apache-2.0
```

yunchu committed:
Updated pre-commit hooks

- Enabled 'ruff' as a base linter
- Fixed some issues reported by 'ruff' and 'mypy'
- Clean up some outdated configurations in pre-commit-config
- Removed unused config files

eunwoosh commented on .pre-commit-config.yaml:
```diff
@@ -96,3 +49,24 @@ repos:
+        args: [--fix]
+
+  # # Git conflict marker
+  # - repo: https://github.com/jumanjihouse/pre-commit-hooks
```
Is this part necessary? If not, can this part be removed?

yunchu replied:
@eunwoosh If you're asking about the --fix option, that would be used for the developer's env to fix some issues that can be fixed automatically while making a commit. and the git conflict marker linting hook will be enabled in the separated PR because it would fix too much files for the whitespace related things.

eunwoosh reviewed (commented):
Thanks for your work! I left a minor comment. Please take a look.

sungmanc commented on otx/algorithms/classification/adapters/mmcls/models/heads/semisl_multilabel_cls_head.py:
```diff
@@ -116,8 +100,8 @@ class SemiMultilabelClsHead:
-        unlabeled_coef (float): unlabeled loss coefficient, default is 1.0
-        use_dynamic_loss_weighting (boolean): whether to use dynamic unlabeled loss weighting, default is True
+        unlabeled_coef (float): unlabeled loss coefficient, default is 1.0.
+        use_dynamic_loss_weighting (boolean): whether to use dynamic unlabeled loss weighting, default is True.
```
```suggestion
        use_dynamic_loss_weighting (boolean): whether to use dynamic unlabeled loss weighting, default is True.
        aux_loss (dict, optional): auxiliary loss function, default is None.
```

sungmanc commented on otx/algorithms/common/adapters/mmcv/hooks/recording_forward_hook.py:
```diff
@@ -16,7 +16,7 @@
 
 from abc import ABC, abstractmethod
-from typing import List, Sequence, Union
+from typing import Sequence, Union
```
```suggestion
from typing import List, Sequence, Union
```

sungmanc commented on otx/algorithms/common/adapters/mmcv/hooks/recording_forward_hook.py:
```diff
@@ -44,7 +44,7 @@ class BaseRecordingForwardHook(ABC):
         self._module = module
         self._handle = None
-        self._records = []  # type: List[torch.Tensor]
+        self._records = []  # type: list[torch.Tensor]
```
```suggestion
        self._records: List[torch.Tensor] = []
```

sungmanc commented on setup.py:
```diff
@@ -105,7 +102,7 @@ def get_extensions():
 
     def _cython_modules():
-        package_root = os.path.dirname(__file__)
+        os.path.dirname(__file__)
```
How about deleting this line?

yunchu replied:
I've focused on the only linting error. but it should be removed. thanks!

sungmanc commented on otx/api/configuration/helper/create.py:
```diff
@@ -37,7 +37,7 @@
 from .utils import deserialize_enum_value, input_to_config_dict
 
-ParameterGroupTypeVar = TypeVar("ParameterGroupTypeVar", bound=ParameterGroup)
+ParameterGroupTypeVar = TypeVar("ParameterGroupTypeVar", ParameterGroup, ConfigurableParameters)
```
Is this also kind of `pylint` or `mypy` issue?

yunchu replied:
@sungmanc It was related with mypy error. I think it's better to check whether this change is valid or correct or not by someone else.

sungmanc reviewed (commented):
Thanks for nice fixing, left some comments

yunchu committed:
Update otx/algorithms/classification/adapters/mmcls/models/heads/semisl_multilabel_cls_head.py

thanks for your finding! but I cannot understand why the missing docstring for the 'aux_loss' was not detected...

Co-authored-by: Sungman Cho <sungman.cho@intel.com>

samet-akcay commented:
Thanks @yunchu for enabling ruff! It provides a significant speedup indeed!

yunchu committed:
Apply suggestions from code review

thanks!

Co-authored-by: Sungman Cho <sungman.cho@intel.com>

samet-akcay commented on .pre-commit-config.yaml:
```diff
@@ -96,3 +49,24 @@ repos:
+    rev: "v0.0.258"
+    hooks:
+      - id: ruff
+        exclude: "tests|otx/recipes"
```
Any plans to include `tests` in the future?

yunchu replied:
I think that should be but not this time ^^;

yunchu committed:
Update setup.py

Removed unnecessary statement.
